### PR TITLE
Update read_reaction_set_data API

### PR DESCRIPTION
**Pull request by pbauman**
This adds a new overload of `read_reaction_set_data` that takes an existing pointer to a `ParserBase` object. Really, we just want to do #260, but I need this right away so that I can leverage the mixture specification in the XML file (which requires the constructor from #254 and is not exposed to the other `read_reaction_set_data` methods). So with this PR, I can construct the XMLParser object myself and just pass to this method. Added unit test of the function that also tests xml file with multiple species and reaction sections in it.

I'd like to merge quickly after a little GRINS testing.

### CODE REVIEW

**Comment by pbauman** [comment]
GRINS is good with this PR.

File: share/antioch_default_files/antioch_default_chemical_mixture.dat [hunk 1]
```diff
@@ -80,6 +80,7 @@ O	16.00000	  1.5420000000e7 	1.5 	 0
 O+	15.99945	  9.7560000000e7 	1.5 	 1
 O2	32.00000	  0.000000000000 	2.5 	 0
 O2+	31.99945	  3.6370000000e7 	2.5 	 1
+O3      48.00000          3.0104166666e6        3.0      0
```
Comment (roystgnr) [inline share/antioch_default_files/antioch_default_chemical_mixture.dat]: Is this also from http://cccbdb.nist.gov/hf0K.asp ?

**Comment by roystgnr** [comment]
:+1: now if you're still consistent with that comment; :+1: after you cite a source otherwise.

**Comment by pbauman** [comment]
lol, just force pushed a small API update to my new API. I'd copy-pasted, but the filename argument in the new function was unnecessary/superfluous. Not sure why compiler didn't throw unused-variable warning.

File: share/antioch_default_files/antioch_default_chemical_mixture.dat (hunk 1 above)
Comment (pbauman) [reply share/antioch_default_files/antioch_default_chemical_mixture.dat]: Correct. I should've included in the commit log that the comment exactly as stated is still correct. I'll force push an amendment to the commit log.

**Comment by pbauman** [comment]
OK, force pushed again with amended commit log on the Antioch default data.

**Pull request merged**